File: 80_Remove_Duplicates_From_Sorted_Array_II.java
/*

Generally if we need to keep no duplicates, then
int i = 0;
for (int number : nums) {
    if (i < 1 || number > nums[i-1])
        nums[i++] = number;
}
return ;

Now if we need 2 duplicates, then
        int i = 0;
        for (int number : nums) {
            if (i < 2 || number > nums[i-2])
                nums[i++] = number;
        }
        return i;
        
Therefore if we need say k duplicates then
    int i = 0;
    for (int number: nums) {
    if (i < k || number > nums[i-k])
        nums[i++] = number;
    }
    return i;
        

*/

class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0;
        for (int number : nums) {
            if (i < 2 || number > nums[i-2])
                nums[i++] = number;
        }
        return i;
    }
}

File: Set1_1000/80_Remove_Duplicates_From_SOrted_Array_II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        for element in nums[::-1]:
            if nums.count(element) > 2:
                nums.remove(element)
        
        return len(nums)



class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        for element in nums[::-1]:
            if nums.count(element) > 2:
                nums.remove(element)
        
        return len(nums)
